src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +142,207 @@ async function uploadWithProgress({
+
+        const fileOption : vscode.QuickPickItem = { 
+            label: addCodiconToString('file', fileNameToDisplay) 
+        }
+        const selectMore: vscode.QuickPickItem = {
+            label: "Browse for more files..."
+        }
+
+        const picker = createQuickPick({
+            options: {
+                canPickMany: false,
+                ignoreFocusOut: true,
+                title: localize('AWS.message.selectFileUpload', 'Select a file to upload'), 
+                step: 1,
+                totalSteps: 2,
+            },
+            items: [fileOption, selectMore],
+            buttons: [vscode.QuickInputButtons.Back]
+        })
+
+        
+        const response = verifySinglePickerOutput(await promptUserFunction({
+            picker: picker,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                }
+            },
+        }))
+
+        if(!response){
+            return undefined
+        }
+
+        if (response.label === 'Browse for more files...') {
+            fileLocation = await promptForFileLocation(window)
+        }
+
+    }
+    
+    if (!fileLocation) {
+        return undefined
+    }
+    return fileLocation
+}
+
+/**
+ * Uploads given file to a bucket
+ * 
+ * @param fileLocation file uri to be uploaded
+ * @param bucket to upload file to
+ * @param fileSizeBytes 
+ * @param window 
+ * @param outputChannel 
+ * 
+ */
+
+export async function uploadFileToS3Command(
+    fileLocation: vscode.Uri,
+    bucket: S3.Bucket,
+    s3Client: S3Client,
+    fileSizeBytes: FileSizeBytes = statFile,
+    window = Window.vscode(),
+    outputChannel = ext.outputChannel,
+): Promise<void>{
+    
+    const fileName = path.basename(fileLocation.fsPath)
+    const key = '' + fileName
+    const destinationPath = readablePath({ bucket: {name: bucket.Name!}, path: key })
+    
+
+    try {
+        showOutputMessage(`Uploading file from ${fileLocation} to ${destinationPath}`, outputChannel)
+        const request = {
+            bucketName: bucket.Name!,
+            key: key,
+            fileLocation: fileLocation,
+            fileSizeBytes: fileSizeBytes(fileLocation),
+            s3Client,
+            window: window
+        }
+
+        await uploadWithProgress(request)
+        showOutputMessage(`Successfully uploaded file ${destinationPath} to ${bucket.Name!}`, outputChannel)
+        telemetry.recordS3UploadObject({ result: 'Succeeded' })
+        recordAwsRefreshExplorer()
+        vscode.commands.executeCommand('aws.refreshAwsExplorer')
+
+    } catch (e) {
+        getLogger().error(`Failed to upload file from ${fileLocation} to ${destinationPath}: %O`, e)
+        showErrorWithLogs(localize('AWS.s3.uploadFile.error.general', 'Failed to upload file {0}', fileName), window)
+        telemetry.recordS3UploadObject({ result: 'Failed' })
```

Comment:
Might be worth removing the telemetry from this function and instead having the control function handle it. Potentially return a boolean depending on success?

Reply:
It would then have more duplicated code in both commands, with extra lines that aren't that needed